clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/producer/KafkaProducerFactory.java

```diff
@@ -46,9 +49,37 @@
      * @param producerName The name to identify the producer.
      * @param config The Kafka configuration with which the producer is to be created.
      * @return an existing or new producer.
+     * @throws org.apache.kafka.common.KafkaException If creating the producer failed because of an invalid config or
+     *             because bootstrap servers could not be resolved.
```

Comment:
do we still need this method?

Reply:
The `getOrCreateProducer` method is still used.
I haven't replaced all occurrences of this method with the new `getOrCreateProducerWithRetries` method.
The `[..]WithRetries` method is now used in places where it is called in a `start()` method. This is the main idea here, that when the component is just starting up, the other cluster components might also not be ready yet so that bootstrap URLs aren't resolvable yet.

For scenarios where the URLs stop being resolvable while the Hono component is already running, the corresponding method invocations will need to use a specific, configurable `retriesTimeout` value (instead of using `UNLIMITED_RETRIES_DURATION`), for example in the `KafkaBasedEventSender` class where sending an event should not be retried indefinitely.
Therefore, more changes would be needed for that. I would rather not do this as part of this PR.